llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -1200,7 +1200,7 @@ void MachineCopyPropagation::BackwardCopyPropagateBlock(
     // Ignore non-trivial COPYs.
     std::optional<DestSourcePair> CopyOperands =
         isCopyInstr(MI, *TII, UseCopyInstr);
-    if (CopyOperands) {
+    if (CopyOperands && MI.getNumImplicitOperands() == 0) {
```

Comment:
I thought isCopyInstr was already filtering out these hard cases

Reply:
Might as well just not use isCopyInstr, you're probably filtering out cases the target code decided are OK

Reply:
For RISC-V, we want to support the canonical copy instruction that appears after ExpandPostRAPseudos, "ADDI rd, rs, 0". What are you suggesting we use here instead of isCopyInstr?